integrations/clickup/clickup/client.py

```diff
@@ -0,0 +1,254 @@
+from typing import Any, AsyncGenerator, List, Dict
+import httpx
+from loguru import logger
+from port_ocean.context import event
+from port_ocean.context.ocean import ocean
+from port_ocean.utils import http_async_client
+
+PAGE_SIZE = 50
+
+class ClickUpClient:
+    def __init__(self, clickup_url: str, clickup_api_key: str) -> None:
+        """
+        Initializes an instance of the ClickUpClient class.
+
+        Args:
+            clickup_url (str): The base URL for the ClickUp API.
+            clickup_api_key (str): The API key for authenticating requests to the ClickUp API.
+        """
+        self.clickup_url = clickup_url
+        self.api_url = f"{self.clickup_url}/api/v2"
+        self.api_key = clickup_api_key
```

Comment:
You don't need to store api_key as class variable since you can just use clickup_api_key in the auth header directly